src/scene_server/topo_server/logics/model/group.go

```diff
@@ -0,0 +1,310 @@
+		metadata.QueryCondition{Condition: cond})
+	if err != nil {
+		blog.Errorf("find group by object but do http request failed, err: %v, cond: %#v, rid: %s", err, cond,
+			kit.Rid)
+		return nil, err
+	}
+
+	return rsp.Info, nil
+}
+
+// UpdateObjectGroup update object group
+func (g *group) UpdateObjectGroup(kit *rest.Kit, cond *metadata.UpdateGroupCondition) error {
+
+	if cond.Data.Name != nil && len(*cond.Data.Name) != 0 {
+		if err := util.ValidModelNameField(*cond.Data.Name, metadata.GroupFieldGroupName, kit.CCError); err != nil {
+			blog.Errorf("failed to valid the group name(%s), rid: %s", cond.Data.Name, kit.Rid)
+			return kit.CCError.Errorf(common.CCErrCommParamsInvalid, metadata.GroupFieldGroupName)
+		}
+	}
+
+	input := metadata.UpdateOption{
+		Condition: mapstr.MapStr{
+			common.BKFieldID: cond.Condition.ID,
+		},
+		Data: mapstr.MapStr{
+			common.BKPropertyGroupNameField: cond.Data.Name,
+			common.BKIsCollapseField:        cond.Data.IsCollapse,
+		},
+	}
+
+	// generate audit log of object attribute group.
+	audit := auditlog.NewAttributeGroupAuditLog(g.clientSet.CoreService())
+	generateAuditParameter := auditlog.NewGenerateAuditCommonParameter(kit,
+		metadata.AuditUpdate).WithUpdateFields(input.Data)
+	auditLog, err := audit.GenerateAuditLog(generateAuditParameter, cond.Condition.ID, nil)
+	if err != nil {
+		blog.Errorf("generate audit log failed before update attribute group, groupName: %s, err: %v, rid: %s",
+			cond.Data.Name, err, kit.Rid)
+		return err
+	}
+
+	// to update.
+	_, err = g.clientSet.CoreService().Model().UpdateAttributeGroupByCondition(kit.Ctx, kit.Header, input)
+	if err != nil {
+		blog.Errorf("update object group failed, err: %v, input: %#v, rid: %s", err, cond, kit.Rid)
+		return err
+	}
+
+	// save audit log.
+	if err := audit.SaveAuditLog(kit, *auditLog); err != nil {
+		blog.Errorf("update object attribute group %s success, but save audit log failed, err: %v, rid: %s",
+			cond.Data.Name, err, kit.Rid)
+		return err
+	}
+	return nil
+}
+
+// UpdateObjectAttributeGroup Update Object Attribute group
+func (g *group) UpdateObjectAttributeGroup(kit *rest.Kit, conds []metadata.PropertyGroupObjectAtt,
+	modelBizID int64) error {
+
+	propertyGroupIDs := make([]string, 0)
+	objIds := make([]string, 0)
+	for _, cond := range conds {
+		propertyGroupIDs = append(propertyGroupIDs, cond.Data.PropertyGroupID)
+		objIds = append(objIds, cond.Condition.ObjectID)
+	}
+	grpCond := mapstr.MapStr{
+		metadata.GroupFieldGroupID: map[string][]string{
+			common.BKDBIN: propertyGroupIDs,
+		},
+		metadata.GroupFieldObjectID: map[string][]string{
+			common.BKDBIN: objIds,
+		},
+	}
+
+	grps, err := g.FindObjectGroup(kit, grpCond, modelBizID)
+	if err != nil {
+		blog.Errorf("update object attribute group failed, err: %v, input: %#v, rid: %s", err, grpCond, kit.Rid)
+		return err
+	}
+
+	grpMap := make(map[string]metadata.Group)
+	for _, grp := range grps {
+		grpMap[grp.GroupID] = grp
+	}
```

Comment:
原本的判断是数量正好为1，可以在上面遍历的时候判断如果grpMap[grp.GroupID]已经存在的话则报错重复，然后下面还是判断是否存在，grpMap可以用map[string]struct{}
```suggestion
	grpMap := make(map[string]struct{})
	for _, grp := range grps {
		if _, ok := grpMap[propertyGroupID]; ok {
			blog.Errorf("there is more than one group (%s), rid: %s", propertyGroupID, kit.Rid)
			return kit.CCError.Errorf(common.CCErrCommParamsInvalid, metadata.GroupFieldGroupID)
		}
		grpMap[grp.GroupID] = struct{}{}
	}
```